integrations/symfony-bundle.rst

```diff
@@ -0,0 +1,176 @@
+
+
+Configure your client
+`````````````````````
+
+You can configure your clients with some good default options. The clients are later registered as services.
+
+.. code-block:: yaml
+
+    httplug:
+        clients:
+            my_guzzle5:
+                factory: 'httplug.factory.guzzle5'
+                config:
+                    # These options are given to Guzzle without validation.
+                    defaults:
+                        base_uri: 'http://google.se/'
+                        verify_ssl: false
+                        timeout: 4
+                        headers:
+                            Content-Type: 'application/json'
+            acme:
+                factory: 'httplug.factory.guzzle6'
+                config:
+                    base_uri: 'http://google.se/'
+
+.. code-block:: php
+
+    $httpClient = $this->container->get('httplug.client.my_guzzle5');
+    $httpClient = $this->container->get('httplug.client.acme');
+
+
+Plugins
+```````
+
+You can configure the clients with plugins.
+
+.. code-block:: yaml
+
+    // services.yml
+    acme_plugin:
+          class: Acme\Plugin\MyCustonPlugin
+          arguments: ["%api_key%"]
+
+.. code-block:: yaml
+
+    // config.yml
+    httpug:
+        plugins:
+            cache:
+                cache_pool: 'my_cache_pool'
+        clients:
+            acme:
+                factory: 'httplug.factory.guzzle6'
+                plugins: ['acme_plugin', 'httplug.plugin.cache', ''httplug.plugin.retry']
+                config:
+                    base_uri: 'http://google.se/'
+
+
+Authentication
+``````````````
+
+.. code-block:: yaml
+
+    // config.yml
+    httpug:
+        plugins:
+            authentication:
+                my_basic:
+                    type: 'basic'
+                    username: 'my_username'
+                    password: 'p4ssw0rd'
+                my_wsse:
+                    type: 'wsse'
+                    username: 'my_username'
+                    password: 'p4ssw0rd'
+                my_brearer:
+                    type: 'bearer'
+                    token: 'authentication_token_hash'
+                my_service:
+                    type: 'service'
+                    service: 'my_authentication_service'
+
+        clients:
+            acme:
+                factory: 'httplug.factory.guzzle6'
+                plugins: ['httplug.plugin.authentication.my_wsse']
+
+
+
+Use for Reusable Bundles
+------------------------
+
+Rather than code against specific HTTP clients, you want to use the Httplug `Client` interface. To avoid building your own infrastructure to define services for the client, simply `require: php-http/httplug-bundle` in your bundles `composer.json`. You SHOULD provide configuration for each of your services that needs an HTTP client to specify the service to use, defaulting to `httplug.client`. This way, the default case needs no additional configuration for your users.
```

Comment:
You SHOULD provide a configuration option to specify the HTTP client service to use for each... This option should default to ``httplug.client``. This way, the default case needs no additional configuration for your users, but they have the option of using specific clients with each of your services.